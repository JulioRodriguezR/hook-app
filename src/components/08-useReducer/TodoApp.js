import React, {useReducer} from 'react';

import { todoReducer } from './todoReducer';
import './styles.css';

const initialState = {
  id: new Date().getTime(),
  desc: 'Learn Redux',
  done: false
};

export const TodoApp = () => {

  const [state] = useReducer(todoReducer, initialState);
  console.log(state);

  return (
    <div>
      <h1>Todo App</h1>
      <hr />

      <ul>
        <li>Hello</li>
        <li>World</li>
      </ul>
    </div>
  )
}
