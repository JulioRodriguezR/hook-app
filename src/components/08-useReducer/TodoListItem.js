import React from 'react'

import './styles.css';

export const TodoListItem = ({ todo, index, handleDelete, handleToggle}) => {

  return (
    <div>
      <li key={todo.id} className="list-group-item">
        <p className={`${todo.done && 'complete'}`} onClick={() => handleToggle(todo.id)}>{index + 1}. {todo.desc}</p>
        <button className="btn btn-danger ml-2" onClick={() => handleDelete(todo.id)}>Remove</button>
      </li>
    </div>
  );

}
