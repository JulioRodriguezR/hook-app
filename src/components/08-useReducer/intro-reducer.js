const initialState = [{
  id: 1,
  todo: 'Todo 1',
  done: false
}];

const todoReducer = (state = initialState, action) => {
  if (action?.type === 'add') {
    return [...state, action.payload];
  }
  return state;
};

let todos = todoReducer();


const newTodo = {
  id: 2,
  todo: 'Todo 2',
  done: false
};

const addTodoAction = {
  type: 'add',
  payload: newTodo // lo que quieres enviar
};


todos = todoReducer(todos, addTodoAction);
console.log(todos);