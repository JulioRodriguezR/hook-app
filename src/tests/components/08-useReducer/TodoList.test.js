import React from 'react';
import { shallow } from 'enzyme';

import { TodoList } from '../../../components/08-useReducer/TodoList';
import { demoTodos } from '../fixtures/demoTodo';

describe('Test in <TodoList />', () => {

  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test('Should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should 2 items TodoList', () => {
    expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
  }); 

});