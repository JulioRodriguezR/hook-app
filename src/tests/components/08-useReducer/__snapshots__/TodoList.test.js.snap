// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <TodoList /> Should show correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Todo 1",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Todo 2",
        "done": true,
        "id": 2,
      }
    }
  />
</ul>
`;
